ui/tests/acceptance/access/identity/_shared-alias-tests.js

```diff
@@ -0,0 +1,79 @@
+import page from 'vault/tests/pages/access/identity/aliases/add';
+import aliasIndexPage from 'vault/tests/pages/access/identity/aliases/index';
+import aliasShowPage from 'vault/tests/pages/access/identity/aliases/show';
+import createItemPage from 'vault/tests/pages/access/identity/create';
+import showItemPage from 'vault/tests/pages/access/identity/show';
+
+export const testAliasCRUD = (name, itemType, assert) => {
+  let itemID;
+  let aliasID;
+  if (itemType === 'groups') {
+    createItemPage.createItem(itemType, 'external');
+  } else {
+    createItemPage.createItem(itemType);
+  }
+  andThen(() => {
+    let idRow = showItemPage.rows.filterBy('hasLabel').filterBy('rowLabel', 'ID')[0];
+    itemID = idRow.rowValue;
+    page.visit({ item_type: itemType, id: itemID });
+  });
+  page.editForm.name(name).submit();
+  andThen(() => {
+    let idRow = aliasShowPage.rows.filterBy('hasLabel').filterBy('rowLabel', 'ID')[0];
+    aliasID = idRow.rowValue;
+    assert.equal(
+      currentRouteName(),
+      'vault.cluster.access.identity.aliases.show',
+      'navigates to the correct route'
+    );
+    assert.ok(
+      aliasShowPage.flashMessage.latestMessage.startsWith('Successfully saved', `${itemType}: shows a flash message`)
+    );
+    assert.ok(aliasShowPage.nameContains(name), `${itemType}: renders the name on the show page`);
+  });
+
+  aliasIndexPage.visit({ item_type: itemType });
+  andThen(() => {
+    assert.equal(aliasIndexPage.items.filterBy('id', aliasID).length, 1, `${itemType}: lists the entity in the entity list`);
+    aliasIndexPage.items.filterBy('id', aliasID)[0].menu();
+  });
+  aliasIndexPage.delete().confirmDelete();
+
+  andThen(() => {
+    assert.equal(aliasIndexPage.items.filterBy('id', aliasID).length, 0, `${itemType}: the row is deleted`);
+    aliasIndexPage.flashMessage.latestMessage.startsWith('Successfully deleted', `${itemType}: shows flash message`);
+  });
+};
+
+export const testAliasDeleteFromForm = (name, itemType, assert) => {
+  let itemID;
+  let aliasID;
+  if (itemType === 'groups') {
+    createItemPage.createItem(itemType, 'external');
+  } else {
+    createItemPage.createItem(itemType);
+  }
```

Comment:
This isn't so bad, but I always get nervous putting control flow in test helpers. It makes them brittle and can be really hard to trace values to.